builder/dockerfile/shell_parser.go

```diff
@@ -40,11 +42,12 @@ func ProcessWord(word string, env []string) (string, error) {
 // this splitting is done **after** the env var substitutions are done.
 // Note, each one is trimmed to remove leading and trailing spaces (unless
 // they are quoted", but ProcessWord retains spaces between words.
-func ProcessWords(word string, env []string) ([]string, error) {
```

Comment:
Can you add a comment here? Something along the lines of:
```
On non-Windows systems the 'escapeToken' should be '\\'. On Windows systems it should be '`'.
```


Just curious, this is always run on the daemon side, right? Can we just check to see whether we're on Windows vs Linux and choose the correct rune value w/o asking for it to be passed in?